src/pip/_internal/cli/req_command.py

```diff
@@ -132,12 +143,26 @@ def make_requirement_preparer(
                     "fast-deps has no effect when used with the legacy resolver."
                 )
 
+        # Handle build constraints
+        build_constraints = getattr(options, "build_constraints", [])
+        constraints = getattr(options, "constraints", [])
+        build_constraint_feature_enabled = (
+            hasattr(options, "features_enabled")
+            and options.features_enabled
+            and "build-constraint" in options.features_enabled
+        )
```

Comment:
`features_enabled` is part of the general option group. It should always exist.
```suggestion
        build_constraint_feature_enabled = "build-constraint" in options.features_enabled
```